src/main/scala/fs2/Process1.scala

```diff
@@ -104,6 +104,12 @@ object process1 {
   def scan1[F[_],I](f: (I, I) => I)(implicit F: NotNothing[F]): Handle[F,I] => Pull[F,I,Handle[F,I]] =
     Pull.receive1 { case o #: h => scan(o)(f).apply(h) }
 
+  /** Writes the sum of all input elements, or zero if the input is empty. */
+  def sum[F[_],I:Numeric](implicit F: NotNothing[F]): Handle[F,I] => Pull[F,I,Handle[F,I]] = {
```

Comment:
That's fine. I was just pointing out that if you do have multiple monoid
libraries you are working with, you can overload functions to work with
any/all of them. If you're only working with one that is fine too. Either
way you are empowered as the user to add new functions that aren't provided
already by FS2, and you don't have to duplicate code. FS2 won't provide
every possible useful function and that is okay. :)

I don't follow what you are asking about in you second paragraph. Just give
the instances like you would for anything else? perhaps in the companion
object if it's a type you control or in a standard place if it's for a
third party library. Same sort of idea as what scalaz does, with instances
it provides for its own types and for the standard library.


On Wed, Nov 4, 2015 at 1:01 AM Jed Wesley-Smith <notifications@github.com>
wrote:

> In src/main/scala/fs2/Process1.scala
> <https://github.com/scalaz/scalaz-stream/pull/473#discussion_r43845631>:
>
> > @@ -104,6 +104,12 @@ object process1 {
> >    def scan1[F[_],I](f: (I, I) => I)(implicit F: NotNothing[F]): Handle[F,I] => Pull[F,I,Handle[F,I]] =
> >      Pull.receive1 { case o #: h => scan(o)(f).apply(h) }
> >
> > +  /** Writes the sum of all input elements, or zero if the input is empty. */
> > +  def sum[F[_],I:Numeric](implicit F: NotNothing[F]): Handle[F,I] => Pull[F,I,Handle[F,I]] = {
>
> as a user I wouldn't do that though, I'd basically only define and use my
> own MonoidalFns.
>
> Think about it though, I'm not sure what you're planning with plugging in
> impls for the fs2.util.* I can't really think of a way that doesn't
> involve orphans. I was assuming that Monoid would use exactly the same
> technique, but need to think further as to how you might achieve it without
> magic.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scalaz/scalaz-stream/pull/473/files#r43845631>.
>